src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -1455,6 +1452,7 @@ bool StatefulWriter::matched_reader_remove(
         return true;
     }
 
+    check_acked_status();
```

Comment:
As we have not modified the list of matched readers, I don't think this is necessary